plotly/figure_factory/_facet_grid.py

```diff
@@ -0,0 +1,1073 @@
+        if 'xaxis' in key or 'yaxis' in key:
+            fig['layout'][key]['tickfont'] = {
+                'color': TICK_COLOR, 'size': 10
+            }
+            fig['layout'][key]['gridcolor'] = GRID_COLOR
+            fig['layout'][key]['gridwidth'] = 2
+            fig['layout'][key]['zeroline'] = False
+
+    # axis titles
+    x_title_annot = _axis_title_annotation(x, 'x')
+    y_title_annot = _axis_title_annotation(y, 'y')
+    fig['layout']['annotations'].append(x_title_annot)
+    fig['layout']['annotations'].append(y_title_annot)
+
+    # legend
+    fig['layout']['showlegend'] = show_legend
+    fig['layout']['legend']['bgcolor'] = LEGEND_COLOR
+    fig['layout']['legend']['borderwidth'] = LEGEND_BORDER_WIDTH
+    fig['layout']['legend']['x'] = 1.05
+    fig['layout']['legend']['y'] = 1
+    fig['layout']['legend']['yanchor'] = 'top'
+
+    if show_legend:
+        legend_annot = _legend_annotation(color_name)
+        fig['layout']['annotations'].append(legend_annot)
+        fig['layout']['margin']['r'] = 150
+
+    # add shaded regions behind axis titles
+    if show_boxes:
+        _add_shapes_to_fig(fig, ANNOT_RECT_COLOR, flipped_rows, flipped_cols)
+
+    axis_labels = {'x': [], 'y': []}
+    for key in fig['layout']:
+        if 'xaxis' in key:
+            axis_labels['x'].append(key)
+        elif 'yaxis' in key:
+            axis_labels['y'].append(key)
+
+    string_number_in_data = False
+    for var in [x, y]:
+        if isinstance(df[var][0], str):
+            for item in df[var]:
+                try:
+                    int(item)
+                    string_number_in_data = True
+                except ValueError:
+                    pass
+
+    if string_number_in_data:
+        for x_y in axis_labels.keys():
+            for axis_name in axis_labels[x_y]:
+                fig['layout'][axis_name]['type'] = 'category'
+
+    if scales == 'fixed':
+        fixed_axes = ['x', 'y']
+    elif scales == 'free_x':
+        fixed_axes = ['y']
+    elif scales == 'free_y':
+        fixed_axes = ['x']
+    else:
+        fixed_axes = []
+
+    # fixed ranges
+    for x_y in ['x', 'y']:
+        min_ranges = []
+        max_ranges = []
+        for trace in fig['data']:
+            if len(trace[x_y]) > 0:
+                min_ranges.append(min(trace[x_y]))
+                max_ranges.append(max(trace[x_y]))
+        while None in min_ranges:
+            min_ranges.remove(None)
+        while None in max_ranges:
+            max_ranges.remove(None)
+
+        min_range = min(min_ranges)
+        max_range = max(max_ranges)
+
+        range_are_numbers = (isinstance(min_range, Number) and
+                             isinstance(max_range, Number))
+
+        if range_are_numbers:
+            # modify ranges
+            min_range = math.floor(min_range)
+            max_range = math.ceil(max_range)
+            if widen_frame:
+                min_range -= 1
+                max_range += 1
```

Comment:
Good call in fixing the range across all subplots so that it's easy to compare values.

I think we can be a bit smarter about setting the autorange though and then make it the default and remove this `widen_frame` argument:
- Adding/subtracting `1` like this won't work if the numerical ranges are really big or if the numerical ranges are within a small range like between `0.5` and `0.6` (in which case it'll end up making the range too large)
- Instead, why don't we just make it say 5% bigger than the absolute range. I believe this is how plotly.js does it. Something like: `range = [min - (max - min) * 0.05, max + (max - min) * 0.05]`. You could ask the #plotly_js folks exactly what value they use